src/services/utils.rs

```diff
@@ -39,21 +43,63 @@ pub async fn estimate_safe_tx_gas(
     safe_transaction_estimation_request: &SafeTransactionEstimationRequest,
 ) -> ApiResult<SafeTransactionEstimation> {
     let info_provider = DefaultInfoProvider::new(chain_id, &context);
-    let url = core_uri!(
+    let estimation_url = core_uri!(
         info_provider,
         "/v1/safes/{}/multisig-transactions/estimations/",
         safe_address
     )?;
+    let latest_multisig_tx_url = core_uri!(
+        info_provider,
+        "/v1/safes/{}/multisig-transactions/?ordering=-nonce&trusted=true&limit=1",
+        safe_address
+    )?;
 
-    let response = context
+    let latest_nonce = fetch_latest_nonce(&context, latest_multisig_tx_url).await?;
+    let safe_tx_gas = fetch_estimation(
+        &context,
+        estimation_url,
+        safe_transaction_estimation_request,
+    )
+    .await?;
+
+    Ok(SafeTransactionEstimation {
+        latest_nonce,
+        safe_tx_gas,
+    })
+}
+
+async fn fetch_estimation(
+    context: &Context<'_>,
+    request_url: String,
+    safe_transaction_estimation_request: &SafeTransactionEstimationRequest,
+) -> ApiResult<String> {
+    let estimation_response = context
         .client()
-        .post(url)
+        .post(request_url)
         .json(safe_transaction_estimation_request)
         .timeout(Duration::from_millis(default_request_timeout()))
         .send()
         .await?;
 
-    Ok(serde_json::from_str::<SafeTransactionEstimation>(
-        &response.text().await?,
-    )?)
+    Ok(serde_json::from_str::<BackendSafeTransactionEstimation>(
+        &estimation_response.text().await?,
+    )?
+    .safe_tx_gas)
+}
+
+async fn fetch_latest_nonce(context: &Context<'_>, request_url: String) -> ApiResult<u64> {
+    let latest_multisig_tx_response = context
+        .client()
+        .get(request_url)
+        .timeout(Duration::from_millis(default_request_timeout()))
+        .send()
+        .await?;
+
+    Ok(serde_json::from_str::<Page<BackendMultisigTransaction>>(
+        &latest_multisig_tx_response.text().await?,
+    )?
+    .results
+    .first()
+    .map(|it| it.nonce)
+    .unwrap_or(0))
```

Comment:
nitpick: The indentation here looks a bit off (maybe because of fmt). The `Ok()` seems to be only applied to the first statement. I was wondering if we could just extract it in this case.

```rust
let nonce = serde_json::from_str::<Page<BackendMultisigTransaction>>(
        &latest_multisig_tx_response.text().await?,
    )?
    .results
    .first()
    .map(|it| it.nonce)
    .unwrap_or(0);

Ok(nonce)
```

Reply:
I agree, `rustfmt` does not look too nice here. I will add your suggestion, it is also stating more cleary what we expect after the `unwrap_or`